### PR TITLE
SW-4370 Added address search hooks

### PR DESCRIPTION
**Pull request by tommylau523**

### CODE REVIEW

**Comment by tommylau523** [comment]
* **#2745** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2745?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/components/Map/EditableMapV2.tsx [hunk 1]
```diff
@@ -320,6 +322,17 @@ export default function EditableMap({
           : {}),
       }}
     >
+      <MapSearchBox
+        onSelect={(features: AddressAutofillFeatureSuggestion[] | null) => {
+          if (features && features.length > 0) {
+            const coordiantes = features[0].geometry.coordinates;
```
Comment (nickgraz) [inline src/components/Map/EditableMapV2.tsx:328]:
```suggestion
            const coordinates = features[0].geometry.coordinates;
```

File: src/utils/useMapboxSearch.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  suggestText?: string;
+};
+
+const useMapboxSearch = (): MapboxSearch => {
+  const { token } = useMapboxToken();
+
+  const addressAutofillCore = new AddressAutofillCore({
+    accessToken: token,
+  });
+  const session = new SearchSession(addressAutofillCore, SUGGESTION_DELAY_MS);
```
Comment (nickgraz) [inline src/utils/useMapboxSearch.ts:31]: Should this and the `session.addEventListener` be inside a `useEffect`? These will fire again any time a component using this hook rerenders, but that may be OK

**Review by nickgraz — APPROVED** [review]
This looks good, one small question about the Mapbox initialization but it may not be an issue.

File: src/utils/useMapboxSearch.ts (hunk 2 above)
Comment (tommylau523) [reply src/utils/useMapboxSearch.ts:31]: Good idea! Doesn't seem like it will hurt.

**Comment by tommylau523** [comment]
### Merge activity

* **Jun 18, 1:21 PM EDT**: @tommylau523 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2745).

**Pull request merged**